iceoryx_hoofs/source/posix_wrapper/file_lock.cpp

```diff
@@ -31,50 +32,68 @@ namespace posix
 {
 constexpr const char FileLock::LOCK_FILE_SUFFIX[];
 
-FileLock::FileLock(const FileName_t& name) noexcept
-    : m_name(name)
-    , m_fileLockPath(platform::IOX_LOCK_FILE_PATH_PREFIX + m_name + LOCK_FILE_SUFFIX)
+cxx::expected<FileLock, FileLockError> FileLockBuilder::create() noexcept
 {
-    initializeFileLock().and_then([this]() { this->m_isInitialized = true; }).or_else([this](FileLockError& error) {
-        this->m_isInitialized = false;
-        this->m_errorValue = error;
-    });
-}
-
-cxx::expected<FileLockError> FileLock::initializeFileLock() noexcept
-{
-    if (!cxx::isValidFilePath(m_name))
+    if (!cxx::isValidFileName(m_name))
     {
+        LogError() << "Unable to create FileLock since the name \"" << m_name << "\" is not a valid file name.";
         return cxx::error<FileLockError>(FileLockError::INVALID_FILE_NAME);
     }
-    constexpr int CREATE_FILE_FOR_READ_WRITE = O_CREAT | O_RDWR;
-    mode_t userReadWriteAccess = S_IRUSR | S_IWUSR;
 
-    auto openCall = posixCall(iox_open)(m_fileLockPath.c_str(), CREATE_FILE_FOR_READ_WRITE, userReadWriteAccess)
-                        .failureReturnValue(ERROR_CODE)
-                        .evaluate()
-                        .and_then([this](auto& r) { this->m_fd = r.value; });
+    if (!cxx::isValidPath(m_path))
+    {
+        LogError() << "Unable to create FileLock since the path \"" << m_path << "\" is not a valid path.";
+        return cxx::error<FileLockError>(FileLockError::INVALID_PATH);
+    }
+
+    FileLock::FilePath_t fileLockPath = m_path;
+
+    if (!cxx::doesEndWithPathSeparator(fileLockPath))
+    {
+        fileLockPath.unsafe_append(iox::platform::IOX_PATH_SEPARATORS[0]);
+    }
+
+    fileLockPath.unsafe_append(m_name);
+    fileLockPath.unsafe_append(FileLock::LOCK_FILE_SUFFIX);
```

Comment:
... and here I'm able to reuse my reasoning so you already payed your dept. The size of `FileLock::LOCK_FILE_SUFFIX` is not part of the size of `FilePath_t` and might not be appended

Reply:
May your 2 minutes be taken again with an additional 50%processing fee. ;)

Reply:
Then you can keep your truncation :laughing: 

Could you add a test for this?

Reply:
> Could you add a test for this?

Please don't. The problem is that I have to create a max path length so I have to create a new directory which has the maximum length with `mkdir` and remove it later with `rmdir`.
Those functions are of course available in Windows with slightly different parameters and are called there `_mkdir` and `_rmdir`.

I could of course some dirty `ifdef`s or implement the c++17 https://en.cppreference.com/w/cpp/filesystem/create_directory and use it here.

Would you be fine when I create an issue for create and remove directory and add a task to implement this unit test when it is available?

Reply:
Okay, that's a good point. I think we can leave it as is for now

Reply:
I created an issue and linked this comment: https://github.com/eclipse-iceoryx/iceoryx/issues/1430